pkg/uuid/uuid.go

```diff
@@ -26,5 +32,11 @@ type RealUUIDGenerator struct {
 }
 
 func (t RealUUIDGenerator) Generate() string {
-	return uuidv1.Now().String()
+	return timeuuid.TimeUUID().String()
+}
+
+func GetTime(id *UUID) *timestamppb.Timestamp {
```

Comment:
Since this relies on a property of the UUID package that is not immediately obvious, maybe it would be good to have a few unit tests that make sure that the UUID package actually behaves in the way we expect it to? Otherwise it would be quite easy for someone to accidentially break functionality, e.g. by switching to a different UUID version.

Reply:
good point, I added a roundtrip test.